deltacat/compute/compactor/model/round_completion_info.py

```diff
@@ -7,32 +7,28 @@
 from deltacat.compute.compactor.model.primary_key_index import \
     PrimaryKeyIndexVersionLocator
 
-from typing import Any, Dict, Optional
+from typing import Any, Dict, Optional, List
 
 
 class RoundCompletionInfo(dict):
```

Comment:
I'm not sure we can completely avoid multi-rounds. It maybe still necessary when we encounter insufficient resource situation. also I think `roundcompletion` is okey for just one-round compaction job, as it's picked up by the same job everyday for the same table/partition, so it's like a daily round.

Reply:
+1 to keeping RoundCompletionInfo for now. Every separate compaction job run can be thought of as a round, and the desirable end state is to still have this job internally separate a single large job into separate rounds that it can complete serially (the only reason we've considered removing it is due to concessions around garbage collection and memory management that we should also fix).

Reply:
If we are thinking of both as a single concern, the API equivalence should also be established which I don't see at the moment. For example, driver needs to be cognizant about the parameters it should pass to when executing multiple rounds vs just another rebase or incremental compaction. My motivation was that it may become complicated to support both as a single concern while there exists simpler ways if we just think of them as separate concerns. This is more of a post Q1 when we start supporting multiple rounds but in the scope of this PR, we could make this naming change to avoid future readers having to search for what the `round` means when there isn't any. We can also table this discussion now and continue it post Q1 when we are actually supporting multiple rounds.

Reply:
How about we keep this for now, and bring it up post Q1 to either refactor this API or have separate drivers/APIs for single-round and multi-round @raghumdani